### ISSUE TITLE
CUDA and RAM Memory crashing with sub 4GB model on T4 GPU instance

### ISSUE THREAD

**Issue by Rakrocks18** [googlecolab/colabtools#4740]
**The Issue**
The instance is crashing whenever I'm trying to pass an inference to 'google/flan-t5-base' or 'google/flan-t5-base' or 'google/flan-t5-large'. The download size for all is less than 5 GB, yet a 12GB RAM and a 15 GB GPU are being overflown. It is throwing an error that either the RAM is full, or I'm trying to allot more than available ono a CUDA device.

**The expected behavior**
***It should run!***

I'm using Chrome

Link to gist
https://gist.github.com/Rakrocks18/b388a4bd100e4b1502d7cce8c66e42d5
https://colab.research.google.com/drive/16WKpIpioQPI3HihgDmig5DsZBGA0Z9-Y?usp=sharing

**Comment by mayankmalik-colab** [comment]
Just checking the error log, it says `torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 16.86 GiB. GPU` .  T4 has 15 GB ram. Try running it on L4 or A100 GPUs , which have higher ram. Also, the size of a model on disk is of compressed state sometimes. When loaded into RAM, the model is decompressed, potentially taking up significantly more ram.

**Comment by Rakrocks18** [comment]
Yes, I do get that T4 doesn't have that much of vRAM. However, the
small/base/large Flan-T5 are less than 1GB. So asking 16.86GB doesn't make
any sense in anyway.

**Comment by cperry-goog** [comment]
This is not something we can resolve as we don't own the models, how they interact with GPU RAM is out of scope. I'd try it on an L4. Sorry.
